### PR TITLE
Ddp 7085 disabling activity block

### PR DESCRIPTION
**Pull request by YuriyPavlovBI**
DDP-7085 Extend API to support disabling activity block

### CODE REVIEW

**Comment by ssettipalli** [comment]
wondering if this is something that can be handled from UI as one of cases. If checkbox checked , disable the edit button and avoid the API changes
@MocanaAtBroad

**Comment by bskinner** [comment]
> wondering if this is something that can be handled from UI as one of cases. If checkbox checked , disable the edit button and avoid the API changes @MocanaAtBroad

How would a front-end client know _that specific checkbox_ is special?

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 1]
```diff
@@ -760,13 +763,17 @@ Block.Visibility:
...
   properties:
     blockGuid:
       type: string
       description: identifier for the block
     shown:
       type: boolean
       description: whether block is shown or not
+    enabled:
+      type: boolean
+      description: whether block is enabled or not
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml]: What does `enabled` mean for a block? Adding a description of what changes with regards to the blocks behavior would be helpful.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiBlockEnabledExpression.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    long insert(long blockId, long expressionId, long revisionId);
+
+    @UseStringTemplateSqlLocator
+    @SqlQuery("queryEnabledExprTextByBlockIdAndRevision")
+    Optional<String> getExpressionText(@Bind("blockId") long blockId,
+                                       @Bind("instanceGuid") String instanceGuid);
+
+    @SqlQuery("select * from block_enabled_expression as be"
+            + " join revision as rev on rev.revision_id = be.revision_id"
+            + " where be.block_id = :blockId and rev.end_date is null")
```
Comment (bskinner) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiBlockEnabledExpression.java:28]: Will there be cases where `end_date` is set to a date in the future, but not `null`?

**Review by ssettipalli — DISMISSED** [review]
looks ok to me

**Review by bskinner — DISMISSED** [review]

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiBlockEnabledExpression.java (hunk 2 above)
Comment (YuriyPavlovBI) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiBlockEnabledExpression.java:28]: I can imagine the situation you mentioned but for some reason it is made the same way for enabled expressions (see JdbiBlockExpression) so we did the same here. Revisions switch at the time we update the study. So, it seems ok for expressions.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 1 above)
Comment (YuriyPavlovBI) [reply pepper-apis/docs/specification/src/components/schemas.yml]: Added short description

**Review by ssettipalli — APPROVED** [review]

**Pull request merged**